client/containers/DashboardSubmissionWorkflow/SubmissionWorkflowStepPicker.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect } from 'react';
+import classNames from 'classnames';
+import { Tab, TabList, useTabState } from 'reakit/Tab';
+
+import { Icon } from 'components';
+
+import {
+	SubmissionWorkflowConfigStep,
+	submissionWorkflowConfigSteps,
+	submissionWorkflowConfigStepLabels,
+} from './types';
+
+require('./submissionWorkflowStepPicker.scss');
+
+type Props = {
+	selectedStep: SubmissionWorkflowConfigStep;
+	onSelectStep: (step: SubmissionWorkflowConfigStep) => unknown;
+	stepCompletions: Record<SubmissionWorkflowConfigStep, boolean>;
+};
+
+const SubmissionWorkflowStepPicker = (props: Props) => {
+	const { selectedStep, onSelectStep, stepCompletions } = props;
+	const tabs = useTabState({ currentId: selectedStep });
+	const { selectedId, select } = tabs;
+	const selectedStepIndex = submissionWorkflowConfigSteps.indexOf(selectedStep);
+
+	useEffect(() => {
+		if (selectedStep !== selectedId) {
+			select(selectedStep);
+		}
+	}, [selectedStep, selectedId, select]);
+
+	return (
+		<TabList
+			{...tabs}
+			className="submission-workflow-step-picker-component"
+			aria-label="Workflow configuration steps"
+		>
+			{Object.entries(submissionWorkflowConfigStepLabels).map(([step, label], index) => {
+				const reached = selectedStepIndex >= index;
+				const passed = selectedStepIndex > index;
+				const incomplete = passed && !stepCompletions[step];
+				return (
```

Comment:
whether this returns just the `Tab` or a wrapped anything, it'll need a `key={step}` property or similar per React's `Warning: Each child in a list should have a unique "key" prop.`